### ISSUE TITLE
Error wrapping InputStream in GZIPInputStream: java.io.EOFException

### ISSUE THREAD

**Issue by pathikrit** [sbt/sbt#2753]
## steps

Tried `sbt ~test`
## problem

```
Error wrapping InputStream in GZIPInputStream: java.io.EOFException
    at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
    at sbt.WrapUsing.open(Using.scala:34)
    at sbt.Using.apply(Using.scala:23)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:810)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:809)
    at sbt.Using.apply(Using.scala:24)
    at sbt.IO$.gzipFileIn(IO.scala:809)
    at sbt.Sync$.readUncaught(Sync.scala:88)
    at sbt.Sync$.readInfo(Sync.scala:84)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:876)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:872)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:235)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.EOFException
    at java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)
    at java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258)
    at java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)
    at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)
    at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
    at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
    at sbt.Using$$anon$1.openImpl(Using.scala:51)
    at sbt.WrapUsing$$anonfun$open$2.apply(Using.scala:34)
    at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
    at sbt.WrapUsing.open(Using.scala:34)
    at sbt.Using.apply(Using.scala:23)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:810)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:809)
    at sbt.Using.apply(Using.scala:24)
    at sbt.IO$.gzipFileIn(IO.scala:809)
    at sbt.Sync$.readUncaught(Sync.scala:88)
    at sbt.Sync$.readInfo(Sync.scala:84)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:876)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:872)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:235)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```
## expectation

Not throw the above exception
## notes

Did `sbt clean` and it worked fine there after.

**Comment by pathikrit** [comment]
Possible dupe of https://github.com/sbt/sbt/issues/253 --> not sure why it was closed without a resolution

**Comment by jbilcke** [comment]
I have the same issue, except I still can't build the project, even after calling `sbt clean`or even deleting the `target` directory myself:

```
$ sbt compile

...

Error wrapping InputStream in GZIPInputStream: java.io.EOFException
    at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
    at sbt.WrapUsing.open(Using.scala:34)
    at sbt.Using.apply(Using.scala:23)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:863)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:862)
    at sbt.Using.apply(Using.scala:24)
    at sbt.IO$.gzipFileIn(IO.scala:862)
    at sbt.Sync$.readUncaught(Sync.scala:88)
    at sbt.Sync$.readInfo(Sync.scala:84)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:922)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:918)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.EOFException
    at java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)
    at java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258)
    at java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)
    at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)
    at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
    at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
    at sbt.Using$$anon$1.openImpl(Using.scala:51)
    at sbt.WrapUsing$$anonfun$open$2.apply(Using.scala:34)
    at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
    at sbt.WrapUsing.open(Using.scala:34)
    at sbt.Using.apply(Using.scala:23)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:863)
    at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:862)
    at sbt.Using.apply(Using.scala:24)
    at sbt.IO$.gzipFileIn(IO.scala:862)
    at sbt.Sync$.readUncaught(Sync.scala:88)
    at sbt.Sync$.readInfo(Sync.scala:84)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
    at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:922)
    at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:918)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
    at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[error] (reco/compile:copyResources) Error wrapping InputStream in GZIPInputStream: java.io.EOFException
```

Scala version: 2.11.8
Sbt version: 0.13.11

**Comment by jbilcke** [comment]
Ok I found out why it didn't work for me, the `sbt clean` wasn't recursive: I've emptied everything and now it works fine.

**Comment by iurisman** [comment]
sbt is scala's weakest link. This and many other bugs make Scala development a chore it doesn't have to be.

**Comment by eed3si9n** [comment]
I ran into this while switching between sbt versions, so I fixed it in #3134.
Could you try 0.13.16-M1 if it's reproducible?

**Comment by iurisman** [comment]
Thanks, Eugene. I've received this in 0.13.9. Will report back if any issues after the upgrade.
